storage/client/queue/src/main/java/com/azure/storage/queue/QueueServiceClientBuilder.java

```diff
@@ -230,11 +233,28 @@ public QueueServiceClientBuilder credential(SharedKeyCredential credential) {
      * @throws NullPointerException If {@code connectionString} is {@code null}.
      */
     public QueueServiceClientBuilder connectionString(String connectionString) {
-        Objects.requireNonNull(connectionString);
         this.sharedKeyCredential = SharedKeyCredential.fromConnectionString(connectionString);
+        getEndPointFromConnectionString(connectionString);
         return this;
     }
 
+    private void getEndPointFromConnectionString(String connectionString) {
+        HashMap<String, String> connectionStringPieces = new HashMap<>();
+        for (String connectionStringPiece : connectionString.split(";")) {
+            String[] kvp = connectionStringPiece.split("=", 2);
+            connectionStringPieces.put(kvp[0].toLowerCase(Locale.ROOT), kvp[1]);
+        }
+        String accountName = connectionStringPieces.get(ACCOUNT_NAME);
+        try {
+            this.endpoint = new URL(String.format("https://%s.queue.core.windows.net", accountName));
+        } catch (MalformedURLException e) {
+            LOGGER.error("There is no valid account for the connection string. "
+                + "Connection String: %s", connectionString);
+            throw new IllegalArgumentException(String.format("There is no valid account for the connection string. "
```

Comment:
The new guideline is to use `logger.logAndThrow()` instead.

Reply:
This will address in another PR.

Reply:
https://github.com/Azure/azure-sdk-for-java/pull/4552